### ISSUE TITLE
SpaceInsideBlockBraces should allow different configuration for non-argument blocks

### ISSUE THREAD

**Issue by ptarjan** [rubocop-hq/rubocop#2908]
We'd like our codebase to have no spaces if there is an argument, but spaces if there isn't one.

I propose adding an option `EnforcedStyleWithoutArguments` (and optionally renaming  `EnforcedStyle` to `EnforcedStyleWithArguments`) to `SpaceInsideBlockBraces`
## Expected behavior

```
# good
foo { bar }
foo {|baz| bar}
```

```
# bad
foo {bar}
foo { |baz| bar }
```
## Actual behavior

With `space`:

```
foo { bar }
foo { |baz| bar }
```

With `no_space`:

```
foo {bar}
foo {|baz| bar}
```
## RuboCop version

Include the output of `rubocop -V`:

```
$ rubocop -V
0.37.2 (using Parser 2.3.0.6, running on ruby 2.1.2 x86_64-darwin15.0)
```

**Comment by garettarrowood** [comment]
It looks like this is already possible with the `SpaceBeforeBlockParameters` config option.

**Comment by pocke** [comment]
> It looks like this is already possible with the SpaceBeforeBlockParameters config option.

The option almost works well, but it adds an offense to `foo {|baz| bar}`. The code is a part of "good" example in "Expected Behaviour".

```ruby
# good
foo { bar }
foo {|baz| bar}

# bad
foo {bar}
foo { |baz| bar }
```

```yaml
Layout/SpaceInsideBlockBraces:
  SpaceBeforeBlockParameters: false
```

```bash
$ rubocop --only SpaceInsideBlockBraces
Inspecting 1 file
C

Offenses:

test.rb:3:15: C: Layout/SpaceInsideBlockBraces: Space missing inside }.
foo {|baz| bar}
              ^
test.rb:6:6: C: Layout/SpaceInsideBlockBraces: Space missing inside {.
foo {bar}
     ^
test.rb:6:9: C: Layout/SpaceInsideBlockBraces: Space missing inside }.
foo {bar}
        ^
test.rb:7:6: C: Layout/SpaceInsideBlockBraces: Space between { and | detected.
foo { |baz| bar }
     ^

1 file inspected, 4 offenses detected
```

However, I think the behaviour is natural. `foo {|baz| bar}` is bad, but `foo {|baz| bar }` looks good.

**Comment by garettarrowood** [comment]
I think this isn't an issue then. Since it was raised, we've also added a lot of good documentation to this cop.  [This specific example is documented here](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/layout/space_inside_block_braces.rb#L77).

Our default is `space` and not `no_space` though, which is why that is an offense. I personally don't feel strongly about either, but I'm not sure there is enough of an outcry to justify a change to the default.
